build.rs

```diff
@@ -287,6 +295,24 @@ fn read_env_var(name: &'static str) -> Result<String, std::env::VarError> {
     std::env::var(name)
 }
 
+// Find executable in PATH with optional extra search directories
+fn find_in_path(path: &Path, extra: Vec<&Path>) -> Option<PathBuf> {
```

Comment:
I understand your point, so may I suggest to modify it such that specific directories are searched *before* checking the PATH instead? This would ensure that target/tools/windows/nasm/nasm.exe always gets picked up even if nasm.exe is also present in the PATH. Requiring a specific copy of a tool within a custom directory is particularly painful for everybody else. I have nasm installed in my CI environment for other dependencies that require it, I just make sure to make it available in my system PATH so they can pick it up.